From b08adb291f3ae0a2626d50eb8abecbc3a0de63d9 Mon Sep 17 00:00:00 2001
From: "almostmatt@google.com" <almostmatt@google.com>
Date: Fri, 21 Apr 2023 14:31:06 -0400
Subject: [PATCH] remove unused var

---
 src/idl_gen_rust.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/idl_gen_rust.cpp b/src/idl_gen_rust.cpp
index 55b8439b..48539d52 100644
--- a/src/idl_gen_rust.cpp
+++ b/src/idl_gen_rust.cpp
@@ -406,7 +406,6 @@ class RustGenerator : public BaseGenerator {
     // example: f(A, D::E)          -> super::D::E
     // does not include leaf object (typically a struct type).
 
-    size_t i = 0;
     std::stringstream stream;
 
     auto s = src->components.begin();
@@ -417,7 +416,6 @@ class RustGenerator : public BaseGenerator {
       if (*s != *d) { break; }
       ++s;
       ++d;
-      ++i;
     }
 
     for (; s != src->components.end(); ++s) { stream << "super::"; }
-- 
2.40.1.606.ga4b1b128d6-goog

