From 300161c1e2e1414fdd8df7ae6f036a6213c3672f Mon Sep 17 00:00:00 2001
From: "drsanta@google.com" <drsanta@google.com>
Date: Thu, 8 Dec 2022 18:52:28 -0500
Subject: [PATCH] fix want write and sprintf deprecation

---
 src/HTTPSocket.h |  2 +-
 src/Socket.h     | 34 ++++++++++++++++++----------------
 2 files changed, 19 insertions(+), 17 deletions(-)

diff --git a/src/HTTPSocket.h b/src/HTTPSocket.h
index 5cc7a7f..8b1ab20 100644
--- a/src/HTTPSocket.h
+++ b/src/HTTPSocket.h
@@ -204,7 +204,7 @@ struct HttpResponse {
 
             static size_t transform(const char *src, char *dst, size_t length, TransformData transformData) {
                 // todo: sprintf is extremely slow
-                int offset = transformData.hasHead ? 0 : std::sprintf(dst, "HTTP/1.1 200 OK\r\nContent-Length: %u\r\n\r\n", (unsigned int) length);
+                int offset = transformData.hasHead ? 0 : std::snprintf(dst, length, "HTTP/1.1 200 OK\r\nContent-Length: %u\r\n\r\n", (unsigned int) length);
                 memcpy(dst + offset, src, length);
                 return length + offset;
             }
diff --git a/src/Socket.h b/src/Socket.h
index 2179ff8..521e798 100644
--- a/src/Socket.h
+++ b/src/Socket.h
@@ -306,24 +306,26 @@ protected:
         if (messageQueue.empty()) {
 
             if (ssl) {
-                sent = SSL_write(ssl, message->data, message->length);
-                if (sent == (ssize_t) message->length) {
-                    wasTransferred = false;
-                    return true;
-                } else if (sent < 0) {
-                    switch (SSL_get_error(ssl, sent)) {
-                    case SSL_ERROR_WANT_READ:
-                        break;
-                    case SSL_ERROR_WANT_WRITE:
-                        if ((getPoll() & UV_WRITABLE) == 0) {
-                            setPoll(getPoll() | UV_WRITABLE);
-                            changePoll(this);
+                /* BEG Patched by firebase-cpp-sdk 00001-fix-want-write-crash.patch */
+                bool continue_loop = true;
+                do {
+                    sent = SSL_write(ssl, message->data, message->length);
+                    if (sent == (ssize_t) message->length) {
+                        wasTransferred = false;
+                        return true;
+                    } else if (sent < 0) {
+                        switch (SSL_get_error(ssl, sent)) {
+                        case SSL_ERROR_WANT_READ:
+                            continue_loop = false;
+                            break;
+                        case SSL_ERROR_WANT_WRITE:
+                            break;
+                        default:
+                            return false;
                         }
-                        break;
-                    default:
-                        return false;
                     }
-                }
+                } while (continue_loop);
+                /* END Patched by firebase-cpp-sdk 00001-fix-want-write-crash.patch */
             } else {
                 sent = ::send(getFd(), message->data, message->length, MSG_NOSIGNAL);
                 if (sent == (ssize_t) message->length) {
-- 
2.39.0.rc1.256.g54fd8350bd-goog

